cni/log/logger_linux.go

```diff
@@ -1,6 +1,18 @@
 package log
 
+import (
+	"errors"
+
+	"go.uber.org/zap/zapcore"
+)
+
 const (
 	// LogPath is the path where log files are stored.
 	LogPath = "/var/log/"
 )
+
+var ErrETWNotSupported = errors.New("ETW is not supported for Linux")
+
+func GetETWCore(_ string, _ zapcore.Level) (zapcore.Core, error) {
+	return nil, ErrETWNotSupported
+}
```

Comment:
It is still a design issue that I can even call `GetETWCore` from Linux code, and that it always returns an error (by design!).
But if we think about this a little more generically, it becomes much cleaner and we can forget about this error.

Abstractly, what you're doing here is joining a platform-specific logging core in to the main core. This could still be config-driven, if it needed to be, but what's important is that from the perspective of the unplatformed code, it should happen for all platforms in exactly the same way. It shouldn't be an error on Linux if I call the same method on Linux as I do on Windows.

If we stop focusing on "ETW" and shift our perspective to "platform cores"...

```suggestion
func JoinPlatformCores(c zapcore.Core) (zapcore.Core, error) {
	return c, nil
}
```

pt. 1
... continued in suggestion in `logger_windows.go`